include/stumpless/error.h

```diff
@@ -30,85 +30,100 @@
 extern "C" {
 #  endif
 
+
 /**
- * An identifier of the types of errors that might be encountered.
- *
- * Note that the same error may be encountered in different contexts with a
- * different ultimate cause. To completely diagnose a problem, you will need
- * to look at more than just the error id.
+ * A macro to implement the enum values && enum names string array.
  */
-enum stumpless_error_id {
-/**
+#define FOREACH_ERROR(ERROR)	\
```

Comment:
rename this to `STUMPLESS_FOREACH_ERROR` so that it does not interfere with the namespace of any files that include this header